Lib/pathlib/_abc.py

```diff
@@ -936,26 +936,25 @@ def on_error(err):
             def on_error(err):
                 raise err
         try:
```

Comment:
Is this outer try/except redundant, given ``on_error`` is passed to ``Path.walk`` and the inner ``unlink`` and ``rmdir`` instances have try/excepts? The only one that doesn't is in when ``Path.is_dir()`` is false.

Reply:
We can get exceptions from `self.rmdir()` and `self.unlink()`, but the size of the `try:` block is a bit gratuitous. I'll see if I can refactor...